django-hosting-heroku/README.md

```diff
@@ -0,0 +1,44 @@
+# How to Host Your Django Project on Heroku
+
+This is a companion project to a tutorial on Real Python. It's a bare-bones Django project without any views.
+
+## Instructions
+
+You should follow the step-by-step instructions in the tutorial to deploy this project to Heroku. Alternatively, you can run the finished project by following the instructions below.
+
+Download the code by cloning the `materials` repository:
+
+```shell
+$ git clone git@github.com:realpython/materials.git
+```
+
+Create a new local Git repository:
+
+```shell
+$ cd materials/django-hosting-heroku/source_code_final/portfolio-project
+$ git init
+$ git add .
+$ git commit -m "Initial commit"
+```
+
+Deploy the code to Heroku:
+
+```shell
+$ heroku login
+$ heroku create
+$ heroku config:set DEBUG=True SECRET_KEY=$(date | base64)
```

Comment:
I'm going to randomly (ha! 🙂) chime in here on how the value for `SECRET_KEY` gets generated. IMO using `date` poses a security risk since an attacker can enumerate possible outputs: 

```
Thu Jun 24 17:14:14 UTC 2021
Thu Jun 24 17:14:15 UTC 2021
Thu Jun 24 17:14:16 UTC 2021
...
```

On Linux and macOS I'd probably use something like this to get a truly random secret:

```console
$ head -c 32 /dev/urandom | base64
TYXP4IjJ15OMxVoQj7kMDyqSWX3Kdp5yMNtNYI7TXl0=

$ openssl rand -base64 32
oKWPHf8My8WfL88yoTgcgC7JUrY12OwQd92zxtc3JT8=
```

More info here: https://unix.stackexchange.com/questions/230673/how-to-generate-a-random-string

If this is addressed in the article: never mind 🙂 (Although it'd be nice if we could give our readers a solution that's viable for a "real"/production system regardless. Hopefully, this article will be quite popular and so chances are that some folks are just going to copy and paste this snippet without being aware of the security ramifications.) Thanks!

Reply:
This is might also be helpful: https://humberto.io/blog/tldr-generate-django-secret-key/

Nice one-liner for generating a random, URL-safe secret string with Python:

```console
$ python -c "import secrets; print(secrets.token_urlsafe())"
WGnlh-ZDPSNok6avqOiLWqHEVjl0r0ilpWivnODI-E4
```